### PR TITLE
Update training infrastructure

### PR DESCRIPTION
**Pull request by anissingh**
## What?
Updated the training infrastructure to:
- Cache models and optimizers at every epoch.
- Highlight training statistics at every epoch end.
- Move global variables to a configuration file.
- Cached statistics to allow for viewing full training curves at the end of continuation training.
- Added trained models and optimizers to the repo.

## Why?
- Without higher amounts of RAM, the OS will kill the process running the training program after training for sufficiently long. This is problematic for training until validation decreases (or in our case, plateaus).
- Caching models and optimizers at every epoch end allow us to choose the model before overfitting occurs, even if we trained for many epochs after the overfitting occurred.
- Having global variables in their own isolated configuration file allows for an easier to understand way of setting hyperparameters, filepaths, etc.
- Adding trained models and optimizers to the repo allows for future authors to analyze the trained networks and potentially continue to train from a certain model, as well as choose a different model for use.

## Extra Comments
I'm not actually sure if adding the trained models and optimizers is a best practice because they may take up unnecessary space in the repo. However, I do think it's important to **at least** have the model and optimizer in use stored, and if this is the case, I feel like having the other models and optimizers stored is important to prevent someone else from having to retrain the CNN for 70 epochs before they can recover the models and optimizers (all results are, in fact, reproducible due to random seed being set and PyTorch randomness being disabled in `config_settings()`, but it will take a long time to retrain).

### CODE REVIEW

**Review by david-yz-liu — COMMENTED** [review]
@anissingh everything looks good, but I am thinking about tradeoffs with what to store in the repository. I think it's reasonable to store the model that we've chosen as the one to actually use, but storing all 70 seems like overkill. I could see updating the documentation (e.g. either README or the module docstring) to describe the methodology explicitly (ran for 70 epochs, looked at the validation curve, etc.). I could even see storing a CSV file with some of the quantitative metrics of the model evaluation. But in terms of supporting reproducibility, given the main script itself can be used to generate all of the models, I think storing all 70 models doesn't provide enough marginal benefit to outweight the repo size costs. What do you think?

**Comment by anissingh** [comment]
> @anissingh everything looks good, but I am thinking about tradeoffs with what to store in the repository. I think it's reasonable to store the model that we've chosen as the one to actually use, but storing all 70 seems like overkill. I could see updating the documentation (e.g. either README or the module docstring) to describe the methodology explicitly (ran for 70 epochs, looked at the validation curve, etc.). I could even see storing a CSV file with some of the quantitative metrics of the model evaluation. But in terms of supporting reproducibility, given the main script itself can be used to generate all of the models, I think storing all 70 models doesn't provide enough marginal benefit to outweight the repo size costs. What do you think?

I agree, I'll create a verbose `README` explaining the process and remove the rest of the files from the repo.

**Review by david-yz-liu — APPROVED** [review]
